### DISCUSSION TITLE
Is it possible to get the number of words seen per the user in the webview ?

### DISCUSSION THREAD

**Discussion by panaC** [edrlab/thorium-reader#1771] [Ideas]
@danielweck ?

**Comment by danielweck** [comment]
this seems like a good question to ask, but it needs some clarifications:
* define “word” (this is a locale-dependent concept)
* define “seen” (i am assuming “as rendered by DOM CSS”, i.e. excluding hidden elements, including CSS-generated content, ... what about accessible screen reader labels?)
* define “webview” (visible scrolling / paginated viewport, or entire document?)
* is measurement done at a given instant snapshot, or over some period of time?

**Comment by panaC** [comment]
be able to count the number of words (withespace splits) currently seeing by the user in the publication webview.

example: at a T snapshot time, how many words the user can counted in the publication.

I'm thinking to calculate an estimate of time/percentage to complete a chapter

**Comment by danielweck** [comment]
> I'm thinking to calculate an estimate of time/percentage to complete a chapter

My recommendation would be to calculate the total number of "words" in the rendered HTML document, then divide by the number of "pages" (either a scroll extent, or a the number of CSS Columns in paginated mode, accounting for two-page spreads). This is a good-enough approximation.
